kasa/experimental/sslaestransport.py

```diff
@@ -335,42 +341,36 @@ async def perform_handshake2(self, local_nonce, server_nonce, pwd_hash) -> None:
         _LOGGER.debug("Handshake2 complete ...")
 
     async def perform_handshake1(self) -> tuple[str, str, str]:
-        """Perform the handshake."""
-        _LOGGER.debug("Will perform handshaking...")
+        """Perform the handshake1."""
+        resp_dict = None
+        if self._username:
+            local_nonce = secrets.token_bytes(8).hex().upper()
+            resp_dict = await self.try_send_handshake1(self._username, local_nonce)
+            resp_dict["error_code"] = SmartErrorCode.SESSION_EXPIRED.value
+
+        # Try the default username. If it fails raise the original error_code
+        if (
+            not resp_dict
+            or (error_code := self._get_response_error(resp_dict))
+            is not SmartErrorCode.INVALID_NONCE
+        ):
```

Comment:
Maybe worth to extract into its own `_try_login_with_default_creds`?

Reply:
Maybe this comment is no longer relevant as I've removed the erroneous forcing of the default creds try.